src/mame/drivers/8080bw.cpp

```diff
@@ -1318,31 +1318,80 @@ void _8080bw_state::cosmicmo(machine_config &config)
 /*                                                     */
 /*******************************************************/
 
+void _8080bw_state::invasion_io_map(address_map &map)
+{
+  map(0x00, 0x00).portr("IN0");
+  map(0x01, 0x01).portr("IN1");
+  map(0x02, 0x02).portr("IN2");
+  map(0x03, 0x03).portr("IN3");
+  /* That's probably wrong. Sidam Invasion uses a totally redesigned hardware layout. To be done later... */
+  map(0x03, 0x03).w("soundboard", FUNC(invaders_audio_device::p1_w));
+  map(0x05, 0x05).w("soundboard", FUNC(invaders_audio_device::p2_w));
+  map(0x06, 0x06).w(m_watchdog, FUNC(watchdog_timer_device::reset_w));
+
+}
+
+
 static INPUT_PORTS_START( invasion )
 	PORT_INCLUDE( sicv )
 
 	PORT_MODIFY("IN1")
-	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_COIN1 )
-	PORT_BIT( 0x80, IP_ACTIVE_HIGH, IPT_COIN2 )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_COIN2 )
 
 	PORT_MODIFY("IN2")
-	/* SW1:5-8 Unused according to manual. */
-	PORT_DIPNAME( 0x03, 0x03, DEF_STR( Lives ) )        PORT_DIPLOCATION("SW1:4,3") /* Default is OFF, OFF (6) */
-	PORT_DIPSETTING(    0x00, "3" )
-	PORT_DIPSETTING(    0x01, "4" )
-	PORT_DIPSETTING(    0x02, "5" )
-	PORT_DIPSETTING(    0x03, "6" )
-	PORT_DIPUNUSED_DIPLOC( 0x04, 0x04, "SW1:8" )        /* Default OFF */
-	PORT_DIPNAME( 0x08, 0x00, DEF_STR( Bonus_Life ) )   PORT_DIPLOCATION("SW1:2") /* Default is ON (2500) */
-	PORT_DIPSETTING(    0x08, "1500" )
-	PORT_DIPSETTING(    0x00, "2500" )
-	/* SW1:5,6,7 Default ON */
 	PORT_DIPNAME( 0x80, 0x80, "Laser Bonus Info" )      PORT_DIPLOCATION("SW1:1") /* Default is OFF (Info on) */
-	PORT_DIPSETTING(    0x00, DEF_STR( Off ) )
-	PORT_DIPSETTING(    0x80, DEF_STR( On ) )
+	PORT_DIPSETTING(    0x00, DEF_STR( On ) )
+	PORT_DIPSETTING(    0x80, DEF_STR( Off ) )
+  PORT_DIPNAME( 0x08, 0x00, DEF_STR( Bonus_Life ) )   PORT_DIPLOCATION("SW1:2") /* Default is ON (2500) */
+  PORT_DIPSETTING(    0x08, "1500" )
+  PORT_DIPSETTING(    0x00, "2500" )
+  PORT_DIPNAME( 0x03, 0x03, DEF_STR( Lives ) )        PORT_DIPLOCATION("SW1:4,3") /* Default is OFF, OFF (6) */
+  PORT_DIPSETTING(    0x00, "3" )
+  PORT_DIPSETTING(    0x01, "4" )
+  PORT_DIPSETTING(    0x02, "5" )
+  PORT_DIPSETTING(    0x03, "6" )
+
+  PORT_MODIFY("IN0")
+  PORT_BIT( 0x70, IP_ACTIVE_HIGH, IPT_CUSTOM ) PORT_CUSTOM_MEMBER(mw8080bw_state, invaders_in1_control_r)  
+  /* SW1:5-8 Unused but mapped to port 0 in hw. */
+  PORT_DIPUNUSED_DIPLOC( 0x80, 0x00, "SW1:5" )
+  PORT_DIPUNUSED_DIPLOC( 0x07, 0x00, "SW1:6,7,8" )
```

Comment:
Please keep the ports in order.

Reply:
I'm sorry but I don't understant what do you mean with keeping ports in order... Have they to be ordered through port values rather than DIPSW number?

Reply:
You’ve got port `IN0` between `IN2` and `IN3` – it makes it harder to find things if they aren’t in order.  Someone could see `IN1` first and assume `IN0` isn’t modified here.

It’s a good idea to code on the assumption that someone will be trying to understand the code while sleep-deprived at 3AM with a deadline to worry about.

Reply:
Undestood... I followed dipswitch order 1,2,3,4 then 5,6,7,8 and I have not seen the port number thing... I'll move it.
Thank you...